remoter-server/src/configureEmbedding.r

```diff
@@ -0,0 +1,200 @@
+#        "methodSettings": {
+#            "umap": {
+#                "minimumDistance": 0.2,
+#                "distanceMetric": "euclidean"
+#            },
+#            "tsne": {
+#                "perplexity": 30,
+#                "learningRate": 200
+#            }
+#        }
+#    },
+#    "clusteringSettings": {
+#        "method": "louvain",
+#        "methodSettings": {
+#            "louvain": {
+#                "resolution": 0.5
+#            }
+#        }
+#    }
+#},
+
+task <- function(scdata, config, task_name, sample_id){
+
+    # Check wheter the filter is set to true or false
+    # Q: Can we disable the configureEmbedding?
+    #if (as.logical(toupper(config$enabled)))
+    scdata.embedding <- run_computeEmbedding(scdata, config)
+    #else
+    #    scdata.embedding <- scdata.embedding
+
+    cells_order <- rownames(scdata.embedding@meta.data)
+
+    # Create a generic pattern with the UMAP coordinates
+    embeddingPreview <- unname(purrr::map2(
+        scdata.embedding@reductions$umap@cell.embeddings[cells_order, 1],
+        scdata.embedding@reductions$umap@cell.embeddings[cells_order, 2],
+        function(x,y) {list("x"=x,"y"=y)}
+        )
+    )
+
+    # Create plot for cellsets
+    embeddingPreviewByCellSets <- purrr::map2(embeddingPreview,
+        unname(scdata.embedding@active.ident[cells_order]),
+        function(x,y){append(x,list("cluster"=paste("Cluster", y, sep = " ")))}
+    )
+    # Adding color for cellsets
+    embeddingPreviewByCellSets <- purrr::map2(embeddingPreviewByCellSets,
+        unname(scdata.embedding@meta.data[cells_order, "color_active_ident"]),
+        function(x,y){append(x,list("col"=y))}
+    )
+
+    # Create plot for samples
+    embeddingPreviewBySamples <- purrr::map2(embeddingPreview,
+        unname(scdata.embedding@meta.data[cells_order, "type"]),
+        function(x,y){append(x,list("sample"=y))}
+    )
+    
+    # Adding color for samples
+    embeddingPreviewBySamples <- purrr::map2(embeddingPreviewBySamples,
+        unname(scdata.embedding@meta.data[cells_order, "color_samples"]),
+        function(x,y){append(x,list("col"=y))}
+    )
+
+    # Create plot for MT-content
+    embeddingPreviewMitochondrialContent <- purrr::map2(embeddingPreview,
+        unname(scdata.embedding@meta.data[cells_order, "percent.mt"]),
+        function(x,y){append(x,list("mt-content"=y))}
+    )
+
+    # Create plot for Doublet-score
+    embeddingPreviewDoubletScore <- purrr::map2(embeddingPreview,
+        unname(scdata.embedding@meta.data[cells_order, "doublet_scores"]),
+        function(x,y){append(x,list("doublet-score"=y))}
+    )
+
+    plots <- list()
+    plots[generate_plotuuid("", task_name, 0)] <- embeddingPreviewByCellSets
+    plots[generate_plotuuid("", task_name, 1)] <- embeddingPreviewBySamples
+    plots[generate_plotuuid("", task_name, 2)] <- embeddingPreviewMitochondrialContent
+    plots[generate_plotuuid("", task_name, 3)] <- embeddingPreviewDoubletScore
```

Comment:
It is becuase this plot is part of the multisample ticket

Reply:
Okay so is this blocked from merge until that goes ahead?

Reply:
No, I think that we can merge now. `generate_plotuuid `is function from "utils.r" that are in the current master branch, but it isn't in my branch because it has been merged to the master after I create this PR.  I have just added the source

Reply:
Okay sounds good, I'm guessing you'll still need to add a `source('utils.r')` as in the other scripts so that you have it in the namespace